snap/local/launcher/install-prompt.py

```diff
@@ -0,0 +1,114 @@
+
+prompt_text = """
+The Assembly3 workbench needs the software <b>py_slvs</b> to work.<br>
+It is not included in this package due to licensing restrictions.<br>
+<br>
+Do you want to download <b>py_slvs</b> and install it now? <br><br>
+If you answer "no", you will not be asked again.
+"""
+
+dontaskagain_file = os.path.expandvars("$HOME/py_slvs_dontaskagain")
+
+class TerminalView(QWidget):
+  def __init__(self, parent=None):
+    super(TerminalView, self).__init__(parent)
+    self.terminal = QWidget(self)
+    layout = QVBoxLayout(self)
+    layout.addWidget(self.terminal)
+    self.setFixedSize(640, 480)
+
+  def exitcode_file(self):
+    return f"/tmp/exitcode{self.winId()}"
+
+  def start_command(self, command, on_finished):
+    self.process = QProcess(self)
+    self.process.finished.connect(on_finished)
+    self.process.start("xterm",[
+      "-into", str(int(self.winId())),
+      "-fa", "Monospace",
+      "-fs", "14",
+      "-e", command + f"; echo $? > {self.exitcode_file()}"
+      ])
+
+  def exit_code(self):
+    with open(self.exitcode_file()) as f:
+      return int(f.read())
+
+class InstallPrompt(QDialog):
+  def __init__(self):
+      super().__init__()
+
+      self.setWindowTitle("Install required software?")
+
+      self.message = QLabel(prompt_text)
+      
+      self.button_box = QDialogButtonBox(QDialogButtonBox.Yes | QDialogButtonBox.No)
+      self.button_box.accepted.connect(self.accept)
+      self.button_box.rejected.connect(self.reject)
+
+      self.layout = QVBoxLayout()
+      self.layout.addWidget(self.message)
+      self.layout.addWidget(self.button_box)
+  
+      self.setLayout(self.layout)
+
+  def hide_prompt(self):
+    self.message.hide()
+    self.button_box.hide()
+
+  def start_install(self):
+    self.terminal = TerminalView()
+    self.terminal.start_command("pip install py_slvs", self.on_pip_finished)
+    self.layout.addWidget(self.terminal)
+
+  def accept(self):
+    self.hide_prompt()
+    self.start_install()
+
+  def reject(self):
+    touch_dontaskagain()
+    sys.exit()
+
+  def on_pip_finished(self):
+    exit_code = self.terminal.exit_code()
+    if exit_code == 0:
+      print("py_slvs was successfully installed")
+    else:
+      print("py_slvs failed to install")
+    sys.exit(exit_code)
+
+def check_py_slvs_installed():
+  has_py_slvs = True
+  try:
+    import py_slvs
+  except:
+    has_py_slvs = False
+
+  if has_py_slvs:
+    print("py_slvs is already installed")
+    sys.exit(0)
+
+def check_dontaskagain():
+  if os.path.isfile(dontaskagain_file):
+    print("not asking again about py_slvs")
+    sys.exit(0)
+
+def touch_dontaskagain():
+  open(dontaskagain_file, "x").close()
+
+if __name__ == "__main__":
+  check_py_slvs_installed()
+  check_dontaskagain()
+
+  app = QApplication(sys.argv)
+
+  window = InstallPrompt()
+  window.show()
+
+  sys.exit(app.exec_())
```

Comment:
Could we add this or a similar dialog directly into the workbench? We could easily drop the xterm and use the python console for the pip output.

Reply:
Yes, I'll try that soon.